### PR TITLE
Oc 217 contributors

### PR DESCRIPTION
**Pull request by mehrdadgit**

### CODE REVIEW

File: src/admin/items/delete.ts [hunk 1]
```diff
@@ -1,6 +1,6 @@
 import { APIGatewayEvent, APIGatewayProxyResult, Context } from 'aws-lambda';
-import { badRequestResponse, successResponse } from '../../common';
-import { db } from '../../databaseConnect';
+import { badRequestResponse } from '../../common';
+import { deleteItm } from '../../items/model';
```
Comment (Dan-Wood) [inline src/admin/items/delete.ts]: Item?

File: src/admin/items/delete.ts [hunk 2]
```diff
@@ -12,27 +12,12 @@ import { db } from '../../databaseConnect';
...
-      query = `
-        DELETE FROM ${process.env.ITEMS_TABLE}
-        WHERE items.id=$1
-      `;
-    await db.any(query, params);
+    const queryString = event.queryStringParameters;
+    const isAdmin = event.path.match(/\/admin\//) ? true : false;
+    const userId = isAdmin ? null : event.requestContext.identity.cognitoAuthenticationProvider.split(':CognitoSignIn:')[1];
+
+    return (await deleteItm(Number(queryString.id), isAdmin, userId));
```
Comment (Dan-Wood) [inline src/admin/items/delete.ts]: Item?

File: src/admin/items/update.ts [hunk 3]
```diff
@@ -121,71 +120,10 @@ export const updateByS3key = async (event: APIGatewayProxyEvent): Promise<APIGat
...
 
-    let message: string = '';
-    if (data.hasOwnProperty('status')) {
-      // Change the s3 level to Private or Public
-      const levelResponse = await changeS3ProtectionLevel(data.s3_key, data.status ? 'public-read' : 'private');
-      if (!levelResponse) {
-        // If we fail to set the level, console log
-        console.log(`Error updating S3 Protection Level for ${data.s3_key}`);
+    const isAdmin = event.path.match(/\/admin\//) ? true : false;
+    const userId = isAdmin ? null : event.requestContext.identity.cognitoAuthenticationProvider.split(':CognitoSignIn:')[1];
```
Comment (Dan-Wood) [inline src/admin/items/update.ts]: isAdmin is just for the path yeah not for is the user is admin?

Whats the purpose of setting userId to null, everyone has a cognito Sub/Username (uuid) .

File: src/collections/model.ts [hunk 4]
```diff
@@ -32,7 +32,7 @@ export const create = async (requestBody, isAdmin: Boolean) => {
     sqlFields.push('created_at', 'updated_at');
     sqlParams.push('now()', 'now()');
 
-    const query = `INSERT INTO ${process.env.COLLECTIONS_TABLE} (${[...sqlFields]}) VALUES (${[...sqlParams]}) RETURNING id;`;
+    const query = `INSERT INTO ${process.env.COLLECTIONS_TABLE} (${sqlFields.join(', ')}) VALUES (${sqlParams.join(', ')}) RETURNING id;`;
```
Comment (Dan-Wood) [inline src/collections/model.ts]: You don't need join here either?

Unless you need that space?


var a = [1,2,3,4]
console.log(\`Hello ${a}\`)


-- Output --
Hello 1,2,3,4

Comment (mehrdadgit) [reply src/collections/model.ts]: I am aware of that, but I would keep join for readability.

File: src/admin/items/update.ts (hunk 3 above)
Comment (mehrdadgit) [reply src/admin/items/update.ts]: I would not worry about that.

File: src/admin/items/delete.ts (hunk 2 above)
Comment (mehrdadgit) [reply src/admin/items/delete.ts]: No Itm because of possible naming conflict with handler

File: src/admin/items/update.ts (hunk 3 above)
Comment (matthewberryman) [reply src/admin/items/update.ts]: It's a needless check though; please remove for computational efficiency

File: src/admin/items/delete.ts (hunk 1 above)
Comment (mehrdadgit) [reply src/admin/items/delete.ts]: Same, possible naming conflict with handler; so it's intentional.

File: src/collections/model.ts (hunk 4 above)
Comment (matthewberryman) [reply src/collections/model.ts]: I find .join much more readable (sorry @Dan-Wood I had spreads for readability ;)

File: src/admin/items/update.ts (hunk 3 above)
Comment (mehrdadgit) [reply src/admin/items/update.ts]: e.g. if there is no authoriser; there wont be a userid so I would keep it for clarity; there is not much efficiency issue here; is a simple check

Comment (matthewberryman) [reply src/admin/items/update.ts]: There will be an authoriser though, so please change.

Comment (mehrdadgit) [reply src/admin/items/update.ts]: OK; I have lost my lcoal branch I guess, I will fix this one after merge on dev

Comment (matthewberryman) [reply src/admin/items/update.ts]: I will revert the revert in dev and then make that change soon.

File: src/admin/items/delete.ts (hunk 1 above)
Comment (Dan-Wood) [reply src/admin/items/delete.ts]: import deleteItem as deleteItemModal in that case?

Thought it was a typo :P

File: src/collections/model.ts (hunk 4 above)
Comment (Dan-Wood) [reply src/collections/model.ts]: Each to their own.

Backticks are wonderful beasts though!

**Pull request merged**